src/legacy/server/logging/log_reporter.js

```diff
@@ -51,18 +52,33 @@ export function getLoggerStream({ events, config }) {
   let dest;
   if (config.dest === 'stdout') {
     dest = process.stdout;
+    logInterceptor.pipe(squeeze).pipe(format).pipe(dest);
   } else {
     dest = writeStr(config.dest, {
       flags: 'a',
       encoding: 'utf8',
     });
 
-    logInterceptor.on('end', () => {
-      dest.end();
-    });
+    if (config.streamErrorHandling) {
+      pipeline(logInterceptor, squeeze, format, dest, onFinished);
```

Comment:
will look into later. but qq:

does pipeline call dest end automatically? or does onfinished get invoked and then we should check if error.code === 'end' and then call dest.end()

Reply:
Yes, pipeline() automatically handles closing the streams once the pipeline is finished or if an error occurs during processing. This automatic cleanup ensures that resources associated with the streams are properly released when they are no longer needed. This behavior helps prevent memory leaks and ensures efficient resource usage in Node.js applications

Reply:
dang. could do you mind opening an issue? i feel like this legacy stuff needs to go and we updated to use the non legacy, or at least be updated to utilize the non-legacy to modern Node APIs.

Reply:
I will open an new issue for it.